### PR TITLE
tests: qa/standalone: Add trap for signals to restore the kernel core pattern

### PR DESCRIPTION
**Pull request by dzafman**

### CODE REVIEW

**Comment by dzafman** [comment]
retest this please

File: qa/run-standalone.sh [hunk 1]
```diff
@@ -24,6 +24,15 @@ else
     COREPATTERN="core.%e.%p.%t"
 fi
 
+function finish() {
+    if [ -n "$precore" ]; then
+        sudo sysctl -w ${KERNCORE}=${precore}
+    fi
+    exit 0
```
Comment (tchaikov) [inline qa/run-standalone.sh]: no need to `exit 0`. and might want to reuse this function at https://github.com/ceph/ceph/pull/17026/files#diff-5e483a2d055861968b99eacaa3688b6fR108.

Comment (dzafman) [inline qa/run-standalone.sh]: @tchaikov With daemons running if I don't exit here weird things happened.  I don't remember exactly what happened, but I tried it with a common function.  I think I had to ^C twice to get the test to stop running.

Comment (tchaikov) [inline qa/run-standalone.sh]: ahh, okay then!

**Pull request merged**